wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/PreferencesService.java

```diff
@@ -0,0 +1,128 @@
+                          "basically it is regex for filtering preferences by names")
+    @ApiResponses({@ApiResponse(code = 200, message = "Preferences successfully fetched"),
+                   @ApiResponse(code = 500, message = "Internal Server Error")})
+    public Map<String, String> find(@ApiParam("Regex for filtering preferences by names, e.g. '.*github.*' " +
+                                              "will return all the preferences which name contains github")
+                                    @QueryParam("filter")
+                                    String filter) throws ServerException {
+        if (filter == null) {
+            return preferencesManager.find(userId());
+        }
+        return preferencesManager.find(userId(), filter);
+    }
+
+    @POST
+    @Consumes(APPLICATION_JSON)
+    @GenerateLink(rel = LINK_REL_PREFERENCES)
+    @RolesAllowed({"user", "temp_user"})
+    @ApiOperation(value = "Saves preferences of logged in user",
+                  notes = "All the existing user's preferences will be override by this method")
+    @ApiResponses({@ApiResponse(code = 204, message = "Preferences successfully saved"),
+                   @ApiResponse(code = 400, message = "Request doesn't contain new preferences"),
+                   @ApiResponse(code = 500, message = "Couldn't save preferences due to internal server error")})
+    public void save(Map<String, String> preferences) throws BadRequestException, ServerException {
+        if (preferences == null) {
+            throw new BadRequestException("Required non-null new preferences");
+        }
+        preferencesManager.save(userId(), preferences);
+    }
+
+    @PUT
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @GenerateLink(rel = LINK_REL_PREFERENCES)
+    @RolesAllowed({"user", "temp_user"})
+    @ApiOperation(value = "Updates preferences of logged in user",
+                  notes = "The merge strategy is used for update, which means that " +
+                          "existing preferences with keys equal to update preference keys will " +
+                          "be replaces with new values, and new preferences will be added")
+    @ApiResponses({@ApiResponse(code = 200, message = "Preferences successfully updated, response contains " +
+                                                      "all the user preferences"),
+                   @ApiResponse(code = 400, message = "Request doesn't contain preferences update"),
+                   @ApiResponse(code = 500, message = "Couldn't update preferences due to internal server error")})
+    public Map<String, String> update(Map<String, String> preferences) throws ServerException, BadRequestException {
+        if (preferences == null) {
+            throw new BadRequestException("Required non-null preferences update");
+        }
+        return preferencesManager.update(userId(), preferences);
+    }
+
```

Comment:
what about a update(String key, String value) and remove(String key) (no bulk mode)